apps/mobile/src/features/token/components/token-details.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import { Balance } from '@/components/balance/balance';
+import { t } from '@lingui/macro';
+
+import { FungibleCryptoAsset, Money } from '@leather.io/models';
+import { Box, Text } from '@leather.io/ui/native';
+import { capitalize } from '@leather.io/utils';
+
+import { getChainLayerFromAssetProtocol } from '../utils/get-chain-layer-from-protocol';
+import { TokenDescription } from './token-description';
+import { TokenDetailsCard, TokenDetailsWrapper } from './token-details-card';
+import { TokenDetailsTable } from './token-details-table';
+import { TokenOverview } from './token-overview';
+import { TokenPriceChange } from './token-price-change';
+
+interface TokenDetailsProps {
+  accountDetails: React.ReactNode;
+  asset: FungibleCryptoAsset;
+  assetDescription: string;
+  availableBalance: Money;
+  price: Money;
+  changePercent: number;
+  quoteBalance: Money;
+  icon: React.ReactNode;
+}
+
+export function TokenDetails({
+  accountDetails,
+  asset,
+  assetDescription,
+  availableBalance,
+  price,
+  changePercent,
+  quoteBalance,
+  icon,
+}: TokenDetailsProps) {
+  return (
+    <TokenDetailsWrapper>
+      <TokenOverview
+        isLoading={false}
+        heading={icon}
+        availableBalance={
+          <Box flexDirection="row" alignItems="center" gap="1">
+            <Balance balance={availableBalance} variant="heading03" />
+            <Text variant="heading03" color="ink.text-subdued">
+              {asset.symbol}
+            </Text>
+          </Box>
+        }
+        quoteBalance={<Balance balance={quoteBalance} variant="label01" isQuoteCurrency />}
+      />
+      {assetDescription && <TokenDescription>{assetDescription}</TokenDescription>}
+
+      <TokenDetailsTable
+        name={`${capitalize(asset.chain)} (${asset.symbol})`}
+        // for Layer design has 'Layer 1 (Bitcoin)' but no other examples. Only showing 'Layer 1' for now based on previous use of getChainLayerFromAssetProtocol
```

Comment:
I need to check with design and get examples of different tokens. The design only has `BTC` and looks like this:

<img width="248" height="117" alt="Screenshot 2025-07-17 at 19 35 54" src="https://github.com/user-attachments/assets/25fec238-bd6b-415a-8ec1-4d22266d766b" />

I will check the copy with design but decided to use a function we already had for now `getChainLayerFromAssetProtocol`
<img width="354" height="192" alt="Screenshot 2025-07-17 at 19 36 21" src="https://github.com/user-attachments/assets/3e46e45b-1e6d-4efb-ad27-7c89c35fc35c" />